packages/tangle/scheduler.go

```diff
@@ -279,29 +280,33 @@ func (s *Scheduler) schedule() *Message {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
+	// cache the access mana
+	manaCache := s.tangle.Options.SchedulerParams.AccessManaMapRetrieverFunc()
+
+	s.updateActiveNodesList(manaCache)
+
 	start := s.buffer.Current()
 	// no messages submitted
 	if start == nil {
 		return nil
 	}
 
-	// cache the access mana retrieval
-	manas := make(map[identity.ID]float64, s.buffer.NumActiveNodes())
 	getCachedMana := func(id identity.ID) float64 {
-		if mana, ok := manas[id]; ok {
+		if mana, ok := manaCache[id]; ok {
```

Comment:
It seems that it would not cause any problems, as `messagelayer.GetAccessMana` is only used in the dashboard, webapi and faucet, which doesn't rely on the hardcoded MinMana. I moved the MinMana to scheduler as you suggested.